src/status_im/ui2/screens/chat/messages/pinned_message.cljs

```diff
@@ -0,0 +1,89 @@
+(ns status-im.ui2.screens.chat.messages.pinned-message
+  (:require [re-frame.core :as re-frame]
+            [status-im.i18n.i18n :as i18n]
+            [quo.design-system.colors :as colors]
+            [status-im.ui.components.react :as react]
+            [quo.core :as quo]
+            [reagent.core :as reagent]
+            [status-im.chat.models.pin-message :as models.pin-message]
+            [status-im.ui.components.list.views :as list]
+            [status-im.utils.handlers :refer [<sub]]
+            [status-im.ui.screens.chat.message.message :as message]
+            [quo.react-native :as rn]))
+
+(def selected-unpin (reagent/atom nil))
+
+(defn render-pin-fn [{:keys [message-id] :as message}
+                     _
+                     _
+                     {:keys [group-chat public? current-public-key space-keeper]}]
+  [react/touchable-without-feedback {:style {:width "100%"}
+                                     :on-press #(reset! selected-unpin message-id)}
+   [rn/view {:style {:flex-direction     :row
+                     :align-items     :center
+                     :justify-content :space-between
+                     :flex            1
+                     :padding-right   20}}
+    [message/chat-message
+     (assoc message
+            :group-chat group-chat
+            :public? public?
+            :current-public-key current-public-key
+            :show-input? false
+            :pinned false
+            :display-username? true
+            :display-photo? true
+            :last-in-group? true
+            :in-popover? false)
+     space-keeper]
+    [rn/view {:style {:position       :absolute
+                      :right       18
+                      :padding-top 4}}
+     [quo/radio {:value (= @selected-unpin message-id)}]]]])
+
+(def list-key-fn #(or (:message-id %) (:value %)))
+
+(defn pinned-messages-limit-list [chat-id]
+  (let [pinned-messages @(re-frame/subscribe [:chats/pinned-sorted-list chat-id])]
```

Comment:
<sub

Reply:
We agreed on using `<sub` instead of `re-frame/subscribe`?

Reply:
yes